osquery/tables/system/python_packages.cpp

```diff
@@ -153,36 +233,33 @@ QueryData genPythonPackagesImpl(QueryContext& context, Logger& logger) {
     }
   }
   for (const auto& key : paths) {
-    genSiteDirectories(key, results, logger);
+    genSiteDirectories(key, results, logger, 0);
+  }
+
+  // Enumerate user installed python packages
+  auto user_paths = getUserPathList(context);
+  for (const auto& user_path : user_paths) {
+    genSiteDirectories(std::get<std::string>(user_path.at("path")),
+                        results,
+                        logger,
+                        std::get<std::int64_t>(user_path.at("user_id")));
   }
 
   if (isPlatform(PlatformType::TYPE_OSX)) {
     for (const auto& dir : kDarwinPythonPath) {
-      std::vector<std::string> versions;
-      if (!listDirectoriesInDirectory(dir, versions, false).ok()) {
-        continue;
-      }
-
-      for (const auto& version : versions) {
-        // macOS will link older versions to 2.6.
-        auto version_path = fs::path(version).parent_path();
-        if (fs::is_symlink(symlink_status(version_path))) {
-          continue;
-        }
-
-        auto complete = version + "lib/python" +
-                        version_path.filename().string() + "/site-packages";
-        genSiteDirectories(complete, results, logger);
+      std::vector<std::string> versions = traverseVersions(dir);
+      for (const auto& site_path : versions) {
+        genSiteDirectories(site_path, results, logger, 0);
       }
     }
   } else if (isPlatform(PlatformType::TYPE_WINDOWS)) {
     // Enumerate any system installed python packages
     auto installPathKey = "HKEY_LOCAL_MACHINE\\" + kWinPythonInstallKey;
-    genWinPythonPackages(installPathKey, results, logger);
+    genWinPythonPackages(installPathKey, results, logger, 0);
 
     // Enumerate any user installed python packages
     installPathKey = "HKEY_USERS\\%\\" + kWinPythonInstallKey;
-    genWinPythonPackages(installPathKey, results, logger);
+    genWinPythonPackages(installPathKey, results, logger, 0);
```

Comment:
Should `uid` be always `0` or should we attempt to match the uid to the user owner of the package?